dpgen2/fp/deepmd.py

```diff
@@ -0,0 +1,207 @@
+        self,
+        conf_frame: dpdata.System,
+        inputs,
+    ):
+        r"""Define how one Deepmd task is prepared.
+
+        Parameters
+        ----------
+        conf_frame : dpdata.System
+            One frame of configuration in the dpdata format.
+        inputs: str or dict
+            This parameter is useless in deepmd.
+        """
+        conf_frame.to("deepmd/npy", deepmd_input_path)
+
+
+class RunDeepmd(RunFp):
+    def input_files(self) -> List[str]:
+        r"""The mandatory input files to run a Deepmd task.
+
+        Returns
+        -------
+        files: List[str]
+            A list of madatory input files names.
+
+        """
+        return [deepmd_input_path]
+
+    def optional_input_files(self) -> List[str]:
+        r"""The optional input files to run a Deepmd task.
+
+        Returns
+        -------
+        files: List[str]
+            A list of optional input files names.
+
+        """
+        return []
+
+    def run_task(
+        self,
+        teacher_model_path: BinaryFileInput,
+        out: str,
+        log: str,
+    ) -> Tuple[str, str]:
+        r"""Defines how one FP task runs
+
+        Parameters
+        ----------
+        command: str
+            The command of running Deepmd task
+        out: str
+            The name of the output data file.
+
+        Returns
+        -------
+        out_name: str
+            The file name of the output data in the dpdata.LabeledSystem format.
+        log_name: str
+            The file name of the log.
+        """
+        log_name = log
+        out_name = out
+
+        dp, type_map_teacher = self._get_dp_model(teacher_model_path)
+
+        # Run deepmd
+        self._dp_infer(dp, type_map_teacher, out_name)
+
+        ret, out, err = run_command(f'echo "job finished!" > {log_name}', shell=True)
+        if ret != 0:
+            raise TransientError(
+                "vasp failed\n", "out msg", out, "\n", "err msg", err, "\n"
+            )
+
+        return out_name, log_name
+
+    def _get_dp_model(self, teacher_model_path: BinaryFileInput):
+        from deepmd.infer import DeepPot  # type: ignore
+
+        teacher_model_path.save_as_file(deepmd_teacher_model)
+        dp = DeepPot(deepmd_teacher_model)
+
+        type_map_teacher = dp.get_type_map()
+
+        os.remove(deepmd_teacher_model)
+        return dp, type_map_teacher
+
+    def _prep_input(self, type_map_teacher, out_name):
+        ss = dpdata.System()
+        ss = ss.from_deepmd_npy(deepmd_input_path)
+        type_map = ss["atom_names"]
+        assert set(type_map).issubset(set(type_map_teacher)), (
+            f"Error: the type map of system ({type_map}) is not subset of "
+            + f"the type map ({type_map_teacher}) of the teacher model."
+        )
+
+        # make sure the order of elements in sys_type_map
+        # is the same as that in type_map_teacher
+        sys_type_map = [ele for ele in type_map_teacher if ele in set(type_map)]
```

Comment:
The system type_map should be load from the `conf_frame` (input of prep_task). Note that the order of elements in this type_map may be different from that of the teacher model.